interpreter/coverage.go

```diff
@@ -0,0 +1,429 @@
+		}
+
+	case *ast.LogStatement:
+		result = append(result, i.instrumentExpression(s.Value)...)
+
+	case *ast.ReturnStatement:
+		result = append(result, i.instrumentExpression(s.ReturnExpression)...)
+
+	case *ast.SetStatement:
+		result = append(result, i.instrumentExpression(s.Value)...)
+
+	case *ast.SwitchStatement:
+		result = append(result, i.instrumentExpression(s.Control.Expression)...)
+		for _, c := range s.Cases {
+			if c.Test != nil {
+				result = append(result, i.instrumentExpression(c.Test)...)
+			}
+		}
+
+	case *ast.SyntheticBase64Statement:
+		result = append(result, i.instrumentExpression(s.Value)...)
+
+	case *ast.SyntheticStatement:
+		result = append(result, i.instrumentExpression(s.Value)...)
+	}
+
+	return result
+}
+
+func (i *Interpreter) instrumentExpression(exp ast.Expression) []ast.Statement {
+	var result []ast.Statement
+
+	switch e := exp.(type) {
+	case *ast.FunctionCallExpression:
+		for _, arg := range e.Arguments {
+			result = append(result, i.instrumentExpression(arg)...)
+		}
+
+	case *ast.GroupedExpression:
+		result = append(result, i.instrumentExpression(e.Right)...)
+
+	case *ast.InfixExpression:
+		result = append(result, i.instrumentExpression(e.Left)...)
+		result = append(result, i.instrumentExpression(e.Right)...)
+
+	case *ast.PrefixExpression:
+		result = append(result, i.instrumentExpression(e.Right)...)
+
+	case *ast.PostfixExpression:
+		result = append(result, i.instrumentExpression(e.Left)...)
+
+	case *ast.IfExpression:
+		result = append(result, i.instrumentIfExpression(e))
+	}
+
+	return result
+}
+
+func (i *Interpreter) instrumentIfExpression(exp *ast.IfExpression) ast.Statement {
+	markAsBranchCovered := createMarkAsBranchCovered(exp.Condition, getExpressionId(exp))
+
+	i.ctx.Tables[BRANCH_COVERAGE].Properties = append(
+		i.ctx.Tables[BRANCH_COVERAGE].Properties,
+		createInitialTableProperty(getExpressionId(exp)+"_true"),
+		createInitialTableProperty(getExpressionId(exp)+"_false"),
+	)
+
+	markAsBranchCovered.Consequence.Statements = append(
+		markAsBranchCovered.Consequence.Statements,
+		i.instrumentExpression(exp.Consequence)...,
+	)
+
+	markAsBranchCovered.Alternative.Consequence.Statements = append(
+		markAsBranchCovered.Alternative.Consequence.Statements,
+		i.instrumentExpression(exp.Alternative)...,
+	)
+
+	return markAsBranchCovered
+}
+
+func createInitialTableProperty(key string) *ast.TableProperty {
+	return &ast.TableProperty{
+		Key: &ast.String{
+			Meta: &ast.Meta{
+				Token: token.Token{Type: token.STRING, Literal: key},
+			},
+			Value: key,
+		},
+		Value: &ast.String{
+			Meta: &ast.Meta{
+				Token: token.Token{Type: token.STRING, Literal: "false"},
+			},
+			Value: "false",
+		},
+	}
+}
+
+func createMarkAsCovered(table, key string) *ast.FunctionCallStatement {
+	return &ast.FunctionCallStatement{
+		Meta: &ast.Meta{Token: token.Null},
+		Function: &ast.Ident{
+			Meta: &ast.Meta{
+				Token: token.Token{Type: token.STRING, Literal: "testing.table_set"},
```

Comment:
I suggest that you can create new testing function for marking converage like `coverage.mark()` but it's okay for this PR. we can do it later!